internal/provider/iam/resource_group.go

```diff
@@ -0,0 +1,225 @@
+	clients "github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func NewGroupResource() resource.Resource {
+	return &resourceGroup{}
+}
+
+type resourceGroup struct {
+	client *clients.Client
+}
+
+func (r *resourceGroup) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+func (r *resourceGroup) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: fmt.Sprintf(`The group resource manages a HCP Group.
+
+The user or service account that is running Terraform when creating an %s resource must have %s on the parent resource; either the project or organization.`,
+			"`hcp_group`", "`roles/admin`"),
+		Attributes: map[string]schema.Attribute{
+			"resource_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The group's unique identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"resource_name": schema.StringAttribute{
+				Computed: true,
+				Description: fmt.Sprintf("The group's resource name in the format `%s`",
+					"iam/organization/<organization_id>/group/<name>"),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"display_name": schema.StringAttribute{
+				Required:    true,
+				Description: "The group's display_name - maximum length of 50 characters",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 50),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "The group's description - maximum length of 300 characters",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(0, 300),
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceGroup) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *clients.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.client = client
+}
+
+type Group struct {
+	ResourceID   types.String `tfsdk:"resource_id"`
+	ResourceName types.String `tfsdk:"resource_name"`
+	DisplayName  types.String `tfsdk:"display_name"`
+	Description  types.String `tfsdk:"description"`
+}
+
+func (r *resourceGroup) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan Group
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	orgID := r.client.Config.OrganizationID
+	parent := fmt.Sprintf("organization/%s", orgID)
+
+	createParams := groups_service.NewGroupsServiceCreateGroupParams()
```

Comment:
**suggestion:** we should be adding the `context.Context` to our API calls here and below to allow Terraform to cancel the API calls.

```suggestion
	createParams := groups_service.NewGroupsServiceCreateGroupParams().WithContext(ctx)
```